v2/tools/generator/internal/codegen/pipeline/create_storage_types.go

```diff
@@ -39,6 +39,20 @@ func CreateStorageTypes() *Stage {
 			// Filter to the types we want to process
 			typesToConvert := state.Definitions().Where(isResourceOrObject).Where(isNotARMType)
 
+			// HACK: include the APIVersion in the storage types package.
+			// really we don't want storage types to have API Version at all,
+			// but it's difficult to remove the GetApiVersion() Function at the moment
+			storageAPIVersions := make(astmodel.TypeNameSet)
+			for _, tdef := range typesToConvert {
+				if rt, ok := astmodel.AsResourceType(tdef.Type()); ok && rt.HasAPIVersion() {
+					storageAPIVersions.Add(rt.APIVersionTypeName())
+				}
+			}
+
+			for name := range storageAPIVersions {
+				typesToConvert.Add(state.Definitions()[name])
+			}
+
```

Comment:
Could this be done by modifying (and renaming) the predicate `isResourceOrObject` instead?

Reply:
Not very easily, it would require another pass. I think it's good to call this out as the HACK that it is 😁 

Ideally we wouldn't have these at all in the storage types, so this is a reminder that we would like to remove them in future.